praw/models/inbox.py

```diff
@@ -25,6 +26,47 @@ def all(self, **generator_kwargs):
         return ListingGenerator(self._reddit, API_PATH['inbox'],
                                 **generator_kwargs)
 
+    def collapse(self, items):
+        """Mark an inbox message as collapsed.
+
+        :param items: A list containing instances of :class:`.Message` or
+            list of t4 message fullnames.
+
+        Requests are batched at 25 items (reddit limit).
+
+        For example, to collapse all unread Messages, try:
+
+        .. code:: python
+
+            from praw.models import Message
+            unread_messages = []
+            for item in reddit.inbox.unread(limit=None):
+                if isinstance(item, Message):
+                    unread_messages.append(item)
+            reddit.inbox.collapse(unread_messages)
+
+        To collapse a list of modmail messages by fullnames, try:
+
+        .. code:: python
+
+            messages = []
+            for message in reddit.subreddit('mod').mod.inbox(limit=None):
+                messages.append(item.fullname)
+            reddit.inbox.collapse(messages)
+
+        .. seealso::
+
+           :meth:`.Message.uncollapse`
+
+        """
+        while items:
+            if isinstance(items[0], Message):
+                data = {'id': ','.join(x.fullname for x in items[:25])}
```

Comment:
For simplicity and consistency with `mark_(un)read`, can we remove the fullname support? If someone has a list of fullnames, they can relatively easily map them to messages?

Alternatively, if you think there's a lot of value, it would be great to update the other methods to have fullname support. I'd suggest making a helper method that takes the list and the path so that the code is more DRY. Also, if we were to go this route, I'd like to not have the assumption that if the first item in a list is a `Message` the next 24 are also `Message`.